Do not link shared library parts with "-pie -fPIE" as it breaks LTO

Changing the linker command from

```
c++ -fPIC -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -pie -fPIE -std=c++11 -fsized-deallocation -Wno-deprecated-declarations -O2 -g -DNDEBUG -flto=auto -Wl,--as-needed -Wl,--no-add-needed -Wl,--as-needed -Wl,-z,relro,-z,now -shared  -o libkeepassx-autotype-xcb.so CMakeFiles/keepassx-autotype-xcb.dir/keepassx-autotype-xcb_autogen/mocs_compilation.cpp.o CMakeFiles/keepassx-autotype-xcb.dir/AutoTypeXCB.cpp.o  ../../libkeepassx_core.a /usr/lib64/libQt5X11Extras.so.5.15.2 /usr/lib64/libX11.so /usr/lib64/libXi.so /usr/lib64/libXtst.so ../../libautotype.a ../../browser/libkeepassxcbrowser.a ../../qrcode/libqrcode.a /usr/lib64/libQt5Svg.so.5.15.2 /usr/lib64/libqrencode.so ../../fdosecrets/libfdosecrets.a /usr/lib64/libQt5Concurrent.so.5.15.2 /usr/lib64/libsodium.so /usr/lib64/libyubikey.so /usr/lib64/libykpers-1.so ../../libzxcvbn.a /usr/lib64/libargon2.so /usr/lib64/libz.so ../../sshagent/libsshagent.a /usr/lib64/libQt5Network.so.5.15.2 ../../keeshare/libkeeshare.a /usr/lib64/libQt5Widgets.so.5.15.2 ../../crypto/ssh/libcrypto_ssh.a -lquazip1-qt5 -lQt5Core /usr/lib64/libQt5DBus.so.5.15.2 -lX11 /usr/lib64/libgcrypt.so /usr/lib64/libgpg-error.so /usr/lib64/libQt5Gui.so.5.15.2 /usr/lib64/libQt5Core.so.5.15.2 
/usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: /tmp/ccuoa53A.ltrans0.ltrans.o: warning: relocation against `_ZN9QHashData11shared_nullE@@Qt_5' in read-only section `.text'
/usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: /tmp/ccuoa53A.ltrans0.ltrans.o: relocation R_X86_64_PC32 against symbol `_ZTV12AutoTypeChar' can not be used when making a shared object; recompile with -fPIC
/usr/lib64/gcc/x86_64-suse-linux/11/../../../../x86_64-suse-linux/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
```

to

```
c++ -fPIC -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -DNDEBUG -fno-common -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -fstack-protector-strong -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Werror=format-security -Wcast-align -std=c++11 -fsized-deallocation -Wno-deprecated-declarations -O2 -g -DNDEBUG -flto=auto -Wl,--as-needed -Wl,--no-add-needed -Wl,--as-needed -Wl,-z,relro,-z,now -shared  -o libkeepassx-autotype-xcb.so CMakeFiles/keepassx-autotype-xcb.dir/keepassx-autotype-xcb_autogen/mocs_compilation.cpp.o CMakeFiles/keepassx-autotype-xcb.dir/AutoTypeXCB.cpp.o  ../../libkeepassx_core.a /usr/lib64/libQt5X11Extras.so.5.15.2 /usr/lib64/libX11.so /usr/lib64/libXi.so /usr/lib64/libXtst.so ../../libautotype.a ../../browser/libkeepassxcbrowser.a ../../qrcode/libqrcode.a /usr/lib64/libQt5Svg.so.5.15.2 /usr/lib64/libqrencode.so ../../fdosecrets/libfdosecrets.a /usr/lib64/libQt5Concurrent.so.5.15.2 /usr/lib64/libsodium.so /usr/lib64/libyubikey.so /usr/lib64/libykpers-1.so ../../libzxcvbn.a /usr/lib64/libargon2.so /usr/lib64/libz.so ../../sshagent/libsshagent.a /usr/lib64/libQt5Network.so.5.15.2 ../../keeshare/libkeeshare.a /usr/lib64/libQt5Widgets.so.5.15.2 ../../crypto/ssh/libcrypto_ssh.a -lquazip1-qt5 -lQt5Core /usr/lib64/libQt5DBus.so.5.15.2 -lX11 /usr/lib64/libgcrypt.so /usr/lib64/libgpg-error.so /usr/lib64/libQt5Gui.so.5.15.2 /usr/lib64/libQt5Core.so.5.15.2
```

makes the linking work. same applies to the autotype test

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,6 +169,9 @@ elseif(KEEPASSXC_DIST_TYPE STREQUAL "Other")
     unset(KEEPASSXC_DIST)
 endif()
 
+# Create position independent code for shared libraries and executables
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
 if("${CMAKE_SIZEOF_VOID_P}" EQUAL "4")
     set(IS_32BIT TRUE)
 endif()
@@ -292,7 +295,6 @@ check_add_gcc_compiler_flag("-Wcast-align")
 
 if(UNIX AND NOT APPLE)
     check_add_gcc_compiler_flag("-Qunused-arguments")
-    check_add_gcc_compiler_flag("-fPIC")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--no-add-needed -Wl,--as-needed -Wl,--no-undefined")
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,relro,-z,now -pie")
     set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,--no-add-needed -Wl,--as-needed")

